diff --git a/Configurations/25-openwrt.conf b/Configurations/25-openwrt.conf
index 8407bf8..4286876 100644
--- a/Configurations/25-openwrt.conf
+++ b/Configurations/25-openwrt.conf
@@ -44,6 +44,9 @@ my %targets = (
     "linux-x86_64-openwrt" => {
         inherit_from    => [ "linux-x86_64", "openwrt" ],
     },
+    "linux-riscv64-openwrt" => {
+        inherit_from    => [ "linux64-riscv64", "openwrt" ],
+    },
 
 ### Basic default option
     "linux-generic32-openwrt" => {
